tests/NgTests/Feed2CatalogTests.cs

```diff
@@ -3,24 +3,290 @@
 
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
+using System.IO.Compression;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
+using System.Security.Cryptography;
 using System.Threading;
 using System.Threading.Tasks;
 using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using NgTests.Data;
 using NgTests.Infrastructure;
+using NuGet.Packaging.Core;
+using NuGet.Services.Metadata.Catalog;
 using NuGet.Services.Metadata.Catalog.Persistence;
+using NuGet.Versioning;
 using Xunit;
 
 namespace NgTests
 {
     public class Feed2CatalogTests
     {
+        private const string _auditRecordDateTimeFormat = "yyyy-MM-ddTHH:mm:ss.FFFZ";
+        private const string _catalogCommitTimestampDateTimeFormat = "yyyy-MM-ddTHH:mm:ss.FFFFFFFZ";
+        private const string _catalogDateTimeFormat = "yyyy-MM-ddTHH:mm:ss.FFFZ";
+        private const string _catalogUrlDateTimeFormat = "yyyy.MM.dd.HH.mm.ss";
         private const string _feedBaseUri = "http://unit.test";
-        private const string FeedUrlSuffix = "&$top=20&$select=Id,NormalizedVersion,Created,LastEdited,Published,LicenseNames,LicenseReportUrl&semVerLevel=2.0.0";
+        private const string _feedUrlSuffix = "&$top=20&$select=Id,NormalizedVersion,Created,LastEdited,Published,LicenseNames,LicenseReportUrl&semVerLevel=2.0.0";
+        private const string _feedUrlDateTimeFormat = "yyyy-MM-ddTHH:mm:ss.fff0000Z";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task RunInternal_WithNoCatalogAndNoActivity_DoesNotCreateCatalog(bool skipCreatedPackagesProcessing)
+        {
+            using (var test = new Test(skipCreatedPackagesProcessing))
```

Comment:
nit: Why not make `Feed2CatalogTests` into `IDisposable` and make the `Test` members on `Feed2CatalogTests`? This will save up to three lines per test and lots of indents.